Automattic/wp-calypso
# Gutenboarding: new design selector step

Resolves https://github.com/Automattic/wp-calypso/issues/40007

This PR adds the new site design step. Since the behavior of the step depends on a new step that's being created, I've added some temporary changes to be removed once the site preview site is available (when you pick a design, it gets a green border to mark it's selected and you can click on "create my site" from here... once the new step is ready, clicking on a design will take you there and you wouldn't be able to create your site directly from this step).
I've removed a ton of code and let this as a pretty simple "grid of pictures" components. Since we don't know how the pages selector will be in the future, and the chances are that it's not going to be integrated with the design picker anymore, it didn't make sense to keep the complexity it introduced.
Also, I stopped using the themes API and I added a json file that will feed the data of the available designs. At least at first, we want  @iamtakashi and @ianstewart to manually curate what we show there (which themes and which images we use to represent them), so a json file with the metadata seems like a way easier method to manually curate that content.  

pinging also @olaolusoga  and @dubielzyk  so they can get crazy about the CSS usage and fix whatever they want (Especially about the image sizes, since it's hard to reflect the prototype without making a mess with the different media query breakpoints, etc) 

How to test:
==========
1. Go to http://calypso.localhost:3000/gutenboarding (or https://calypso.live/gutenboarding?branch=Update/gutenboarding-design )
2. Go past the first step, adding a vertical and a site title
3. Once you go to the next step, you should see the new design picker.

![image](https://user-images.githubusercontent.com/1554855/76218562-b728e500-6214-11ea-9b8b-17bd096a3b9f.png)
